src/views/InvolvementProfile/components/Membership/index.jsx

```diff
@@ -19,8 +19,10 @@ const Membership = ({ isAdmin, isSiteAdmin, involvementDescription, toggleIsAdmi
   const { involvementCode, sessionCode } = useParams();
   const { profile } = useUser();
 
+  const [isMemberAndNonGuest, setIsMemberAndNonGuest] = useState(false);
```

Comment:
This variable name is misleading because it is also true if the user is a SiteAdmin.

Maybe `canViewAllMembers` would be better, since that's what the value is being used for?

Reply:
Well the value is also used for the join and subscribe button display, that is why I didnt use that name

Reply:
That's not entirely accurate. It's checked to decide if we should display the `MemberList` component, and the logic is that, if we don't show the `MemberList` component then we should show the join and subscribe buttons.

Reply:
So I still think the name `canViewAllMembers` is accurate.

Reply:
I prefer shouldShowMemberships as I think it makes more sense in that it surrounds a conditional state on the component rather than a condition on the user.